cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1377,7 +1377,7 @@ gdv_int32 ascii_utf8(const char* data, gdv_int32 data_len) {
   if (data_len == 0) {
     return 0;
   }
-  return static_cast<gdv_int32>(data[0]);
+  return static_cast<gdv_int32>(static_cast<signed char>(data[0]));
```

Comment:
should we try changing the method signature to be const signed char * insead?

Reply:
It won't build then as "some text" is const char and compiler has some strict validations for that. I have already tried.